### ISSUE TITLE
Do not fire RustSelfConventionInspection for Copyable items

### ISSUE THREAD

**Issue by mkaput** [intellij-rust/intellij-rust#515]
Sample ([source](https://github.com/rust-lang-nursery/regex/blob/master/src/input.rs#L242-L292))

``` rust
#[derive(Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
pub struct Char(u32);

impl Char {
    /// Returns true iff the character is absent.
    #[inline]
    pub fn is_none(self) -> bool { self.0 == u32::MAX }
}
```

triggers RustSelfConventionInspection, while that `self` is totally ok here

**Comment by matklad** [comment]
I saw a number of false positives for this inspection (code in Cargo or Rust stdlib). I wonder if the majority of those were indeed caused by not considering `Copy`.
